tensorflow/python/data/experimental/kernel_tests/data_service_ops_test.py

```diff
@@ -527,6 +528,39 @@ def testDistributeInvalidProcessingMode(self):
           data_service_ops.distribute(
               processing_mode="invalid", service="grpc://localhost:5000"))
 
+  @combinations.generate(test_base.eager_only_combinations())
+  def testZipDifferentProcessingModesDatasets(self):
+    cluster = self.create_cluster(num_workers=1)
+    num_elements = 100
+    ds1 = dataset_ops.Dataset.range(num_elements)
+    ds1 = self.make_distributed_dataset(
+        ds1, cluster, processing_mode="distributed_epoch")
+    ds2 = dataset_ops.Dataset.range(num_elements)
+    ds2 = self.make_distributed_dataset(
+        ds2, cluster, processing_mode="parallel_epochs")
+    ds = dataset_ops.Dataset.zip((ds1, ds2))
+    self.assertDatasetProduces(
+        ds, list(zip(range(num_elements), range(num_elements))),
+        assert_items_equal=True)
+
+  @combinations.generate(test_base.eager_only_combinations())
+  def testZipDifferentProcessingModesDatasetsSharedJobName(self):
+    cluster = self.create_cluster(num_workers=1)
+    num_elements = 100
+    ds1 = dataset_ops.Dataset.range(num_elements)
+    ds1 = self.make_distributed_dataset(
+        ds1, cluster, processing_mode="distributed_epoch", job_name="job_name")
+    ds2 = dataset_ops.Dataset.range(num_elements)
+    ds2 = self.make_distributed_dataset(
+        ds2, cluster, processing_mode="parallel_epochs", job_name="job_name")
+    ds = dataset_ops.Dataset.zip((ds1, ds2))
+    with self.assertRaisesRegex(errors.FailedPreconditionError,
+                                "but there is already an existing job"):
+
+      self.assertDatasetProduces(
```

Comment:
Remove the assert and just call `self.getDatasetOutput(ds)`. The assert doesn't get run anyway since iterating raises an error

Reply:
Sure. Addressed in the commit.